ledfx/dedupequeue.py

```
@@ -0,0 +1,29 @@
+    Event,
+)
+
+class VisDeduplicateQ(asyncio.Queue):
+    def __init__(self, maxsize=0):
+        super().__init__(maxsize)
+
+    def put_nowait(self, item):
+        # Check if is a visualisation update message
+        if item.get("event_type") == Event.DEVICE_UPDATE or item.get("event_type") == Event.VISUALISATION_UPDATE:
+            # check if it is a duplicate and just return without queing if it is
+            if any(self.is_similar(item, existing_item) for existing_item in self._queue):
```

Ruff

Use a single `if` statement instead of nested `if` statements

(SIM102)